### PR TITLE
Fix: vtex search params support

### PR DESCRIPTION
**Pull request by IncognitaDev**
fix: priceFrom facet on Legacy
fix: productClusterSearchableIds facet on Legacy

add: PS params support for product count control

### CODE REVIEW

File: vtex/loaders/intelligentSearch/productListingPage.ts [hunk 1]
```diff
@@ -285,6 +286,8 @@ const loader = async (
 
   const searchArgs = { ...args, query: args.query || pathQuery };
 
+  console.log("IS", pathQuery, searchArgs);
```
Comment (lucis) [inline vtex/loaders/intelligentSearch/productListingPage.ts:289]:
```suggestion
```

File: vtex/loaders/intelligentSearch/productListingPage.ts [hunk 2]
```diff
@@ -310,6 +314,8 @@ const loader = async (
     ).then((res) => res.json()),
   ]);
 
+  console.log("IS", selectedFacets);
```
Comment (lucis) [inline vtex/loaders/intelligentSearch/productListingPage.ts:317]:
```suggestion
```

File: vtex/loaders/intelligentSearch/productListingPage.ts [hunk 3]
```diff
@@ -203,7 +204,7 @@ const queryFromPathname = (
   const isPage = Boolean(pageTypes.length);
   const isValidPathSearch = pathList.length == 1;
   if (!isPage && !isInSeachFormat && isValidPathSearch) {
-    return pathList[0];
+    return decodeURI(pathList[0]);
```
Comment (lucis) [inline vtex/loaders/intelligentSearch/productListingPage.ts:207]: Can you clarify why we need this here?

A code comment would be nice

File: vtex/loaders/legacy/productListingPage.ts [hunk 4]
```diff
@@ -94,14 +94,27 @@ const getTerm = (path: string, map: string) => {
   const mapSegments = map.split(",");
   const pathSegments = removeForwardSlash(path).split("/");
 
-  return pathSegments.slice(0, mapSegments.length).join("/");
+  const term = pathSegments.slice(0, mapSegments.length).join("/");
+  console.log(mapSegments);
```
Comment (lucis) [inline vtex/loaders/legacy/productListingPage.ts:98]:
```suggestion
```

File: vtex/loaders/legacy/productListingPage.ts [hunk 5]
```diff
@@ -94,14 +94,27 @@ const getTerm = (path: string, map: string) => {
...
+  if (mapSegments.includes("priceFrom")) {
+    return term.replace(/de-\d+[,]?[\d]+-a-\d+[,]?[\d]+/, (match) => {
+      return match.replaceAll(",", ".");
+    });
+  }
```
Comment (lucis) [inline vtex/loaders/legacy/productListingPage.ts:103]: Can you create named variables to better explain the algorithm here?

Ex:

const `termWithoutWhatever = term.replace(/de-\d...`

(I can't know what this is doing)

File: vtex/loaders/legacy/productListingPage.ts [hunk 6]
```diff
@@ -111,7 +124,7 @@ const getTermFallback = (url: URL, pageTypes: PageType[], fq: string[]) => {
    */
   const isOneTermOnly = pathList.length == 1;
 
-  if (!pageTypes.length && !fq.length && isOneTermOnly) {
+  if (!pageTypes.length && !map && !fq.length && isOneTermOnly) {
```
Comment (lucis) [inline vtex/loaders/legacy/productListingPage.ts:127]: Can you create named variables to better explain the conditionals here?

File: vtex/loaders/legacy/productListingPage.ts [hunk 7]
```diff
@@ -153,25 +169,27 @@ const loader = async (
...
 
-  const missingParams = typeof maybeMap !== "string" || !maybeTerm;
-  const [map, term] = missingParams
-    ? getMapAndTerm(pageTypes)
-    : [maybeMap, maybeTerm];
-
   const fmap = url.searchParams.get("fmap") ?? map;
   const args = { map, _from, _to, O, ft, fq };
 
+  console.log(args, getTerm(term, map));
```
Comment (lucis) [inline vtex/loaders/legacy/productListingPage.ts:191]:
```suggestion
```

File: vtex/utils/transform.ts [hunk 8]
```diff
@@ -607,6 +618,24 @@ export const filtersToSearchParams = (
...
+  priceFrom: "price",
+  productClusterSearchableIds: "productClusterIds",
+};
+
+export const legacyFacetsNormalize = (map: string, path: string) => {
+  const value = path.replace(
+    /de-(?<from>\d+[,]?[\d]+)-a-(?<to>\d+[,]?[\d]+)/,
+    (_match, from, to) => {
+      return `${from.replace(",", ".")}:${to.replace(",", ".")}`;
+    },
```
Comment (lucis) [inline vtex/utils/transform.ts:631]: Same thing about explaining what's going on.

Comment (IncognitaDev) [reply vtex/utils/transform.ts:634]: Fixed

File: vtex/loaders/legacy/productListingPage.ts (hunk 6 above)
Comment (IncognitaDev) [reply vtex/loaders/legacy/productListingPage.ts]: Fixed

File: vtex/loaders/legacy/productListingPage.ts (hunk 5 above)
Comment (IncognitaDev) [reply vtex/loaders/legacy/productListingPage.ts]: Fixed

File: vtex/loaders/intelligentSearch/productListingPage.ts (hunk 3 above)
Comment (IncognitaDev) [reply vtex/loaders/intelligentSearch/productListingPage.ts:209]: Fixed

**Review by lucis — APPROVED** [review]

**Review by lucis — APPROVED** [review]

**Pull request merged**